django_mongodb_backend/compiler.py

```diff
@@ -548,10 +550,25 @@ def get_combinator_queries(self):
 
     def get_lookup_pipeline(self):
         result = []
+        where = self.get_where()
+        promote_filters = defaultdict(list)
```

Comment:
So `promote_filters` are used to create `pushed_expression`. Is there a reason to use two different terms (promote, pushed)? I know there is some usage of "promote" in Django's SQL generation. My understanding is that Django uses the term "join promotion" to describe the conversion of left outer join to inner join, where possible.

I don't know whether it makes any difference to use "promote" or "push" here, but I tentatively propose `pushed_filters` and `pushed_filter_expression` and will make that edit in my language review, if okay.